### PR TITLE
Remove Buf_io from the HTTP server

### PR DESCRIPTION
**Pull request by robhoes**
Cleaning up the attic... It turned out that the `Buf_io` module, while used in the type signature of every HTTP handler, was not used for anything useful anymore.

Step by step rationale in the commit messages, so best reviewed commit by commit.

### CODE REVIEW

**Review by lindig — APPROVED** [review]
If this compiles that pretty convincing that the code is not needed.

**Comment by robhoes** [comment]
>If this compiles that is pretty convincing that the code is not needed.

Well, I did change bits... it's not just deleting code :)

**Comment by lindig** [comment]
Real pirates ship

**Comment by robhoes** [comment]
Tests have passed and connecting with XenCenter also looks normal.

File: ocaml/libs/http-lib/http_svr.ml [hunk 1]
```diff
@@ -713,78 +713,6 @@ let read_body ?limit req bio =
       else
         Buf_io.really_input_buf ~timeout:Buf_io.infinite_timeout bio length
 
-module Chunked = struct
```
Comment (psafont) [inline ocaml/libs/http-lib/http_svr.ml:716]: Oof, this is used in vhd-tool. with its won copy of the code, of course.

Comment (robhoes) [reply ocaml/libs/http-lib/http_svr.ml:716]: Ah is it a copy of http_svr... I did see the name, but thought it was something else.

Comment (robhoes) [reply ocaml/libs/http-lib/http_svr.ml:716]: The implementation of `ocaml/vhd-tool/src/chunked.ml` looks different. Crucially, it doesn't use `Buf_io`.

**Review by psafont — APPROVED** [review]

**Pull request merged**